test/main/modules.js

```diff
@@ -454,4 +454,38 @@ QUnit.module('QUnit.module', function () {
       }, /Cannot add beforeEach hook outside the containing module/);
     });
   });
+
+  QUnit.module('disallowed async module callbacks', function () {
+    var errorFromThenableCallbackModule;
+    try {
+      QUnit.module('with thenable callback', function () {
+        return { then: function () {} };
+      });
+    } catch (e) {
+      errorFromThenableCallbackModule = e;
+    }
+
+    QUnit.test('module with thenable callback function errored', function (assert) {
+      assert.true(errorFromThenableCallbackModule instanceof Error);
+      assert.strictEqual(errorFromThenableCallbackModule.message, 'Returning a promise from a module callback is not supported. Instead, use hooks for async behavior.');
+    });
+
+    var errorFromPromiseCallbackModule;
+    try {
+      QUnit.module('with promise callback', function () {
+        // eslint-disable-next-line compat/compat
+        return Promise.resolve();
```

Comment:
`Promise` isn't available with IE11 which triggered this lint. Also with the es5 setup I can't use `async` functions in this file.

Reply:
We can use the then-able function return here I think? The added fixture is fine for extra coverage, but I meant it originally for here.

Reply:
👍 I'll remove this one and keep the thenable version